go/consensus/tendermint/api/context.go

```diff
@@ -222,6 +225,17 @@ func (c *Context) WithCallerAddress(callerAddress staking.Address) *Context {
 	return child
 }
 
+// WithSimulation creates a child context in simulation mode.
```

Comment:
> Nit: Would it make sense to extract the context and simulation related changes from the "Add staking runtime messages" commit into a separate commit?

I don't think it would add much value. The PR is already an exemplar case on how to structure things into commits.